packages/backend/src/core/assets/NativeAssetUpdater.ts

```diff
@@ -0,0 +1,109 @@
+import { Logger } from '@l2beat/shared'
+import { ChainId, Hash256, UnixTime, ValueType } from '@l2beat/shared-pure'
+import { setTimeout } from 'timers/promises'
+
+import {
+  ReportRecord,
+  ReportRepository,
+} from '../../peripherals/database/ReportRepository'
+import { ReportStatusRepository } from '../../peripherals/database/ReportStatusRepository'
+import { Clock } from '../Clock'
+import { PriceUpdater } from '../PriceUpdater'
+import { TaskQueue } from '../queue/TaskQueue'
+import {
+  addArbTokenReport,
+  ARBITRUM_PROJECT_ID,
+} from '../reports/custom/arbitrum'
+import {
+  addOpTokenReport,
+  OPTIMISM_PROJECT_ID,
+} from '../reports/custom/optimism'
+import { getReportConfigHash } from '../reports/getReportConfigHash'
+import { ReportProject } from '../reports/ReportProject'
+
+export class NativeAssetUpdater {
+  private readonly configHash: Hash256
+  private readonly taskQueue: TaskQueue<UnixTime>
+  private readonly knownSet = new Set<number>()
+
+  constructor(
+    private readonly priceUpdater: PriceUpdater,
+    private readonly reportRepository: ReportRepository,
+    private readonly reportStatusRepository: ReportStatusRepository,
+    private readonly clock: Clock,
+    private readonly projects: ReportProject[],
```

Comment:
IMO we don't need it, probably the config hash should be a hardcoded value to have a full control over when to recalculate